api/application.go

```diff
@@ -135,29 +134,15 @@ func (h ApplicationHandler) Create(ctx *gin.Context) {
 func (h ApplicationHandler) Delete(ctx *gin.Context) {
 	id := h.pk(ctx)
 	m := &model.Application{}
-	err := h.DB.Transaction(func(tx *gorm.DB) (err error) {
-		db := tx.Preload(clause.Associations)
-		result := db.First(m, id)
-		if result.Error != nil {
-			err = result.Error
-			return
-		}
-		for _, task := range m.Tasks {
-			result := tx.Delete(&task)
-			if result.Error != nil {
-				err = result.Error
-				return
-			}
-		}
-		result = db.Select(clause.Associations).Delete(m)
-		if result.Error != nil {
-			err = result.Error
-			return
-		}
```

Comment:
The Application associates can be deleted like other models because we no longer rely on the Task (delete) hook to delete the bucket.